### PR TITLE
Parametric Spool Holder

### PR DESCRIPTION
**Pull request by henrikssn**
* [x] The mod, firmware configuration or slicer profile is in the correct category
    folder. Printable mods go to `printer_mods/`, firmware configurations
    go to `firmware_configurations/`, slicer profiles go to `slicer_profiles/`.
    Create a subfolder with your name, and place the mods in a subfolder with
    a descriptive name within that folder, e.g.: `/printer_mods/FHeilmann/flux_capacitor`
  * [x] Folders names MUST NOT contain spaces. If possible, make sure file names also 
    do not contain any spaces.
  * [x] For each mod, add a small `README.md` file to its folder with a short description
    of what the mod accomplishes. This readme can be used to add pictures, give assembly
    instructions or specify a bill of materials if the mod requires additional hardware.
  * [x] The PR modifies the top-level `README.md` of the category folder adding the 
    contribution to the table. Read the top part of the file for instructions on how
    to do this. Please preserve the alphabetical ordering while adding new rows. Make sure
    to fill out the compatibility matrix to indicate which versions of the Voron printer
    the submission is compatible with.
  * [x] The mod/configuration/profile has been tested by the person submitting the mod 
    and/or other Voron users. Make sure to add information about how the mod was tested below. 
  * [x] The mod is not merely a slight modification of an official Voron part, configuration 
    or profile (i.e. an official Voron part with a few mm added or removed or a slicer profile 
    which only modifies a few values). *(When in doubt, contact one of the admins in the 
    Voron discord before submitting the PR)*
  * [x] Submitted STLs are printable without support. *(If the mod does not meet this criterion
    join the Voron discord and ask the other users for advice on how to modify the mod such 
    that it does not require supports)*
  * [x] Submitted STL files are not corrupt. *(This can be tested by opening the STL in PrusaSlicer
    and checking if mesh errors are reported.)*
  * [x] Submitted STL files are oriented and scaled properly for printing.
  * [x] Submitted firmware configs or slicer profiles contain no sensitive data (e.g. API keys).

#### Which mods/configurations/profiles are added by this PR?

Parametric Spool Holder

#### How was it tested? 

Printed and used with a 2.5kg ESun ABS+ spool (125mm length version)

### CODE REVIEW

**Review by KiloQubit — COMMENTED** [review]
Thanks for your mod!

I feel like this falls under

> The mod is not merely a slight modification of an official Voron part, configuration
or profile (i.e. an official Voron part with a few mm added or removed or a slicer profile
which only modifies a few values).

It's pretty easy to modify the length of the spoolholder even without it being parametric.

**Comment by henrikssn** [comment]
> Thanks for your mod!

Thanks for your feedback!

> 
> I feel like this falls under
> 
> > The mod is not merely a slight modification of an official Voron part, configuration
> > or profile (i.e. an official Voron part with a few mm added or removed or a slicer profile
> > which only modifies a few values).
> 
> It's pretty easy to modify the length of the spoolholder even without it being parametric.

I also thought so at first, but then I found this non-parametric version already in the repo: https://github.com/VoronDesign/VoronUsers/tree/master/printer_mods/raymondh/LargeSpoolHolders

Btw, it does take quite a while to replicate this from the stock holder as you need to split it in five parts, extrude 20 surfaces, move all the parts into their new places and combine them. I tried that first time before I realized that I needed another 5mm width...

File: printer_mods/README.md [hunk 1]
```diff
@@ -122,6 +122,7 @@ like so:
 | | [Voron2.4_Pins_Mod](./hartk1213/Voron2.4_Pins_Mod) | This mod replaces all the M5 screws for all idlers/pulleys/bearings with 5mm pins  | :x:| :x: | :heavy_check_mark: |:x:|
 | | [Voron2.4_Y_Endstop_Relocation](./hartk1213/Voron2.4_Y_Endstop_Relocation) | This mod relocates the Y endstop to the top A drive piece, to work with Arkeets MGN12 mod  | :x:| :x: | :heavy_check_mark: |:x:|
 | henrikssn | [Plug Panel Mod for Generic Unfiltered Modules](./henrikssn/plug_panel_generic) | v2.4 plug panel for generic unfiltered modules with 66mm x 27mm cutout such as the Delta 06A2 and Qualtek 761-18 | :x: | :x: | :heavy_check_mark: | :x:|
+| | [Parametric Spool Holder](./henrikssn/parametric_spool_holder) | Spool holder where the length is a parameter. | :x: | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark:|
```
Comment (KiloQubit) [inline printer_mods/README.md:125]: Is this compatible with a Switchwire, which has 3030 vertical extrusions?  You have a checkmark in the Switchwire column.  You should probably have an X in there, and instead have a checkmark in the Trident column.

**Review by KiloQubit — CHANGES_REQUESTED** [review]
Ok, thanks for the comment.  Just one minor fix below.

File: printer_mods/README.md (hunk 1 above)
Comment (henrikssn) [reply printer_mods/README.md]: Good point, done

**Review by KiloQubit — APPROVED** [review]
Thanks for the update!

**Pull request merged**